### DISCUSSION TITLE
a second task to allow running with different rules

### DISCUSSION THREAD

**Discussion by xenoterracide** [tbroyer/gradle-errorprone-plugin#53] [General]
I could probably create a task myself somehow, though not sure how (off the top of my head). but would be useful to have a 2nd task that could be run with a different configuration.

use case, sometimes I create variables to allow debugging, but I have no unused variables in my configuration, this means that I can't even build to debug unless I use the variable somehow. It's nice for CI to ship, but not so much if I'm running in a debugger. So I figure if I had a 2nd run in CI that checked this that'd be good.

**Comment by tbroyer** [comment]
You'll find an example of custom compile task in the README.

I wouldn't use that approach though, as that means compiling your code twice; I'd rather reconfigure/disable ErrorProne depending on a project property (or by manually editing your build script when you need it):

```kotlin
if (hasProperty("disableErrorProne")) {
  tasks.withType<JavaCompile>().configureEach {
    options.errorprone.isEnabled.set(false)
  }
}
```

**Comment by xenoterracide** [comment]
yeah, I could conditionally do it somehow maybe, it's just an IDE issue, maybe Intellij sets something, but if it doesn't would be easier to have a second task. Hmm... intellij doesn't run "check" maybe I can disable it from running unless check is being run

**Comment by tbroyer** [comment]
IntelliJ sets the `idea.active` system property (at least when importing projects, probably also when running tasks), so `if (Boolean.getBoolean("idea.active")) { /* we're in IDEA */ }`

**Reply by xenoterracide** [reply]
thanks, it appears I can't close this, I don't know why.

**Reply by tbroyer** [reply]
Yeah, I wanted to try "discussions", this looked like a good candidate. I don't really know either how they work :wink:

**Reply by xenoterracide** [reply]
oh, I didn't know it had moved from being a bug.
